### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by leader2one**

### CODE REVIEW

File: python-work/readme.md [hunk 1]
```diff
@@ -1 +1,15 @@
+### helper.py 
+this file is for getting data
 
+### Find the most liked text review and app_ID
+here I estimated the likes as helpfullness numerator. So I counted them grouped by app_id. I estimated review as a summary column
```
Comment (Akvaby) [inline python-work/readme.md:5]: fix typo

File: python-work/readme.md [hunk 2]
```diff
@@ -1 +1,15 @@
...
+
+### Find the shortest interval between ratings of one user (among all users) + the length of both messages from this interval.
+
+### Find the application for which the most useless comment is written.
+I assumed that the difference between denominator and numerator is the indicator of helpfullness. Estimated text column as a comment
+### Calculate the number of records that cannot be processed.
+I assumed that if some of the id columns are NULL then we cant process these record
+
+### Find the most popular words in positive and negative reviews.
+I estimated summary column as review
```
Comment (Akvaby) [inline python-work/readme.md:15]: In the readme should be information about your program.
You should describe the functionality and all steps that should be done to receive the result. 
Just imagine that you see this repo first time in your life. What you should do, step by step, to make this program work.

File: python-work/records that can't be processed.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import helper
+pop = helper.data()
```
Comment (Akvaby) [inline python-work/records that can't be processed.py:2]: PEP 8

File: python-work/popular words.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import helper
+
+pop = helper.data()
+from collections import Counter
+
```
Comment (Akvaby) [inline python-work/popular words.py:5]: PEP 8

File: python-work/popular words.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+# iterate through rows for text column and if score > 3 then append after transform func to positive else negative
+# transform func - takes string and return then as a list after lowering words and replacing non-letters
+# col 8 is summary
```
Comment (Akvaby) [inline python-work/popular words.py:8]: Probably you can use docstring here

File: python-work/most useless comment.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import helper
+pop = helper.data()
+m = min(pop[row][10] for row in range(1, len(pop)))
```
Comment (Akvaby) [inline python-work/most useless comment.py:3]: please choose meaningful naming for variable

File: python-work/find average mark for each app.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import helper
+pop = helper.data()
+
+dic = {}
```
Comment (Akvaby) [inline python-work/find average mark for each app.py:4]: meaningful naming for variable

File: python-work/find average mark for each app.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+for row in range(1, len(pop)):
+    productId = pop[row][1]
+    score = pop[row][6]
+    if productId in dic.keys():
+        dic[productId].append(score)
+    else:
+        dic[productId] = [score]
+
+ans = [['App ID', 'Average Mark']]
```
Comment (Akvaby) [inline python-work/find average mark for each app.py:14]: meaningful naming for variable

File: python-work/Find the most liked text review and app_ID.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import helper
+pop = helper.data()
+
+m = max(pop[row][4] for row in range(1, len(pop)))
```
Comment (Akvaby) [inline python-work/Find the most liked text review and app_ID.py:4]: meaningful naming for variable

File: python-work/numbers of users voted.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import helper
+pop = helper.data()
+
+dic = {}
```
Comment (Akvaby) [inline python-work/numbers of users voted.py:4]: meaningful naming for variable

File: python-work/popular words.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    for word in review.lower().split():
+        word = word.replace(".", "")
+        word = word.replace(",", "")
+        word = word.replace(":", "")
+        word = word.replace("\"", "")
+        word = word.replace("!", "")
+        if par == 'positive':
+            positive.append(word)
+        else:
+            negative.append(word)
```
Comment (Akvaby) [inline python-work/popular words.py:24]: here you update lists outside the function, I think it is not a good idea. You can read information about the namespace.

File: python-work/shortest interval.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+dic = {}
+
+for row in range(1, len(pop)):
+    user = pop[row][2]
+    time = pop[row][7]
+    if user in dic:
+        dic[user].append(time)
+    else:
+        dic[user] = [time]
+mn = 9999999999
```
Comment (Akvaby) [inline python-work/shortest interval.py:15]: meaningful naming for variable

File: python-work/shortest interval.py [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+for user in dic.keys():
+    dic[user] = sorted(dic[user])
+    times = dic[user]
+    if len(times) > 1:
+        mn = min(mn, times[1]-times[0])
```
Comment (Akvaby) [inline python-work/shortest interval.py:20]: meaningful naming for variable

File: python-work/shortest interval.py (hunk 12 above)
Comment (Akvaby) [inline python-work/shortest interval.py:15]: meaningful naming for variable

File: python-work/helper.py [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import csv
+
+def data():
```
Comment (Akvaby) [inline python-work/helper.py:3]: if you pass file name as a parameter, it will make your function more universal

File: python-work/helper.py [hunk 15]
```diff
@@ -0,0 +1,25 @@
+import csv
+
+def data():
+    pop = []
```
Comment (Akvaby) [inline python-work/helper.py:4]: meaningful naming for variable

File: python-work/find average mark for each app.py (hunk 8 above)
Comment (leader2one) [reply python-work/find average mark for each app.py:16]: which variable?

File: python-work/records that can't be processed.py [hunk 16]
```diff
@@ -1,12 +1,24 @@
...
-    productId = pop[row][1]
-    userId = pop[row][2]
-    id = pop[row][0]
-    time = pop[row][7]
-
-    if productId == None or userId == None or id == None or time == None:
-        cnt += 1
-print(cnt)
\ No newline at end of file
+
```
Comment (Akvaby) [inline python-work/records that can't be processed.py:2]: program didn't match with name of the file

**Review by Akvaby — APPROVED** [review]

**Pull request merged**